docs/sharding.md

```diff
@@ -101,7 +101,98 @@ namespace SimpleChatV3
 }
 ```
 
-For other languages such as JavaScript, we will support route algorithm customization in the future.
+> The dependency of customized negotiation router: 
+> 
+> If you need to implement [`GetNegotiateEndpoint(HttpContext context, IEnumerable<ServiceEndpoint> endpoints)`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR/EndpointRouters/IEndpointRouter.cs) method yourself and rely on the parameter `HttpContext`, for example, use `HttpContext.Request.Query["endpoint"]` to select a nearer endpoint for router, you can only use the HTTP trigger in your negotiation functions, so that your router can get the `HttpContext` object correctly.
+
+#### Other languages
+
+For languages other than C#, we support specifying target endpoints in each request. You will use new binding types to get endpoint information.
+##### Client routing
+The `SignalRConnectionInfo` binding selects one endpoint according th the default routing rule. If you want to customize routing rule, you should use `SignalRNegotiation` binding instead of `SignalRConnectionInfo` binding. 
 
-### The dependency of customized negotiation router
-If you need to implement [`GetNegotiateEndpoint(HttpContext context, IEnumerable<ServiceEndpoint> endpoints)`](https://github.com/Azure/azure-signalr/blob/dev/src/Microsoft.Azure.SignalR/EndpointRouters/IEndpointRouter.cs) method yourself and rely on the parameter `HttpContext`, for example, use `HttpContext.Request.Query["endpoint"]` to select a nearer endpoint for router, you can only use the HTTP trigger in your negotiation functions, so that your router can get the `HttpContext` object correctly.
+`SignalRNegotiation` binding data is the same as `SignalRConnectionInfo`. Here's binding data in a `function.json` file sample:
+```json
+{
+    "type": "signalRNegotiation",
+    "name": "negotiationContext",
+    "hubName": "chat",
+    "direction": "in"
+}
+```
+
+You could also add other binding data such as `userId`, `idToken` and `claimTypeList` just like `SignalRConnectionInfo`.
+
+The return data of `SignalRNegotiation` binding is in the following format:
```

Comment:
*The return data* sounds a little confusing, like something customer need to return. How about simply describe like this.

`SignalRNegotiation` input binding object you can get will be in the following format: